single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/svg/CustomLabelProvider.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright (c) 2025, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.sld.svg;
+
+import com.powsybl.sld.layout.LayoutParameters;
+import com.powsybl.sld.library.SldComponentLibrary;
+import com.powsybl.sld.model.coordinate.Direction;
+import com.powsybl.sld.model.nodes.*;
+import com.powsybl.sld.model.nodes.feeders.FeederWithSides;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+import static com.powsybl.sld.model.coordinate.Direction.*;
+
+/**
+ * Enables the configuration of content displayed in the SLD, for labels and feeder infos.
+ * <p>
+ * Customizations are defined in the constructor's map parameters.
+ *
+ * <p>
+ * The labels map defines what will be displayed for the equipments, and it is indexed by the equipment ID.
+ * The custom content is declared through an SldCustomLabels record: the label will be displayed in the diagram at its standard position;
+ * The additionalLabel will be displayed on the equipment's right side.
+ *
+ * <p>
+ * The feederInfosData map defines what will be displayed along the feeder, and it is indexed by the equipment ID (and a not-null side,
+ * for feeders with sides such as lines and transformers), through the SldFeederContext record;
+ * The custom content is declared via as a list of SldCustomFeederInfos records: the componentType is the info component type name;
+ * labelDirection determines the direction (e.g., IN or OUT for the arrows); label is the string displayed next to the info component.
+ *
+ * @author Christian Biasuzzi {@literal <christian.biasuzzi at soft.it>}
+ */
+public class CustomLabelProvider extends AbstractLabelProvider {
+
+    private static final double LABEL2_OFFSET = 6d;
+
+    public record SldCustomLabels(String label, String additionalLabel) {
```

Comment:
As we're inside single-line-diagram module, don't you think we could get rid of the Sld prefix?

Reply:
You're right. Prefix removed.